app/assets/javascripts/discourse/tests/integration/components/site-setting-test.js

```diff
@@ -88,4 +88,16 @@ module("Integration | Component | site-setting", function (hooks) {
       "jpg, jpeg, png, gif, heic, heif, webp, avif, svg"
     );
   });
+
+  test("prevents decimal in integer setting input", async function (assert) {
+    this.set("setting", {
+      setting: "suggested_topics_unread_max_days_old",
+      value: "90",
+      type: "integer",
+    });
+
+    await render(hbs`<SiteSetting @setting={{this.setting}} />`);
+    await fillIn(query(".input-setting-integer"), "90.5");
```

Comment:
Is `fillIn` "typing" the characters one by one or setting the `value` of the input? 🤔

If it's the former, then I would expect the dot to be "swallowed" and the result to be 905.

If it's the latter, are we really testing your change or just the behaviour provided by html when using "step=1"?

Reply:
[`fillIn`](https://github.com/emberjs/ember-test-helpers/blob/master/API.md#fillin) sets the value all at once.
[`typeIn`](https://github.com/emberjs/ember-test-helpers/blob/master/API.md#typein) is emulating all the steps.
I believe we should use `typeIn` here.

Reply:
Oh yes you are both right, I will use typeIn, I didn't realise this existed and have previously been messing around with our triggerKeyPress code